packages/requisitions/src/ResponseRequisition/DetailView/columns.ts

```diff
@@ -1,45 +1,83 @@
+import { useEffect } from 'react';
 import {
   useColumns,
   GenericColumnKey,
   SortBy,
-  Column,
   ColumnAlign,
+  zustand,
+  useSortBy,
 } from '@openmsupply-client/common';
 import { ResponseLineFragment } from './../api';
 
-interface UseResponseColumnOptions {
+type Store = {
   sortBy: SortBy<ResponseLineFragment>;
-  onChangeSortBy: (
-    column: Column<ResponseLineFragment>
-  ) => SortBy<ResponseLineFragment>;
-}
+  setSortBy: (sortBy: SortBy<ResponseLineFragment>) => void;
+};
 
-export const useResponseColumns = ({
-  onChangeSortBy,
-  sortBy,
-}: UseResponseColumnOptions): Column<ResponseLineFragment>[] =>
-  useColumns<ResponseLineFragment>(
+const useStore = zustand<Store>(set => ({
+  sortBy: { key: 'itemName', isDesc: false, direction: 'asc' },
+  setSortBy: (sortBy: SortBy<ResponseLineFragment>) =>
+    set(state => ({ ...state, sortBy })),
+}));
+
+const useSharedSortBy = () => {
+  const sharedSortBy = useStore();
+  const { sortBy, onChangeSortBy } = useSortBy<ResponseLineFragment>(
+    sharedSortBy.sortBy
+  );
+
+  useEffect(() => {
+    sharedSortBy.setSortBy(sortBy);
+  }, [sortBy]);
+  return { sortBy, onChangeSortBy };
```

Comment:
Tricky one!

1. Render data with columns [from content area hook]
1. Trigger sort by change [from lines hook]
2. Changes hook specific state [lines hook state]
3. Effect changes zustand state [global]
4. returns hook specific state [lines hook state]
5. Lines hook reruns as zustand state changed 
6. Lines hook see the line hook specific state which hasn't changed
7. data doesn't re-render as the columns and data haven't changed

Can return zustand state instead
`{ sortBy: sharedSortBy.sortBy }`

which seems to be working (but I like what you've done moving the hook, seems nice)

Reply:
🎉  thanks! I've shifted to use `sharedSortBy.sortBy` and that works nicely. now the `useColumns` hook can be used safely 😊